storm-core/test/jvm/backtype/storm/nimbus/InMemoryTopologyAcitonNotifier.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.nimbus;
+
+import java.util.Map;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.HashMap;
+
+public class InMemoryTopologyAcitonNotifier implements  ITopologyActionNotifierPlugin {
+
+    //static to ensure eventhough the class is created using reflection we can still get
+    //the topology to actions
+    private static final Map<String, LinkedList<String>> topologyToActions = new HashMap<>();
+
+
+    @Override
+    public void prepare(Map StormConf) {
+        //no-op
+    }
+
+    @Override
+    public synchronized void notify(String topologyName, String action) {
```

Comment:
This is not being invoked in critical (tuple execution) path so I think we don't really care if this method is sync. Do you anticipate huge latencies here?